gcloud/datastore/query.py

```diff
@@ -347,10 +342,10 @@ class Iterator(object):
         datastore_pb.QueryResultBatch.MORE_RESULTS_AFTER_LIMIT,
     )
 
-    def __init__(self, query, connection, limit=None, offset=0,
+    def __init__(self, query, client, limit=None, offset=0,
```

Comment:
Cool. I'll document the constructor args in a follow-up PR.

The value in `fetch` is essentially an override, but it doesn't really matter if the override happens in the caller of the non-public constructor or in the constructor itself.